addons/mail/static/src/discuss/core/common/attachment_panel.xml

```diff
@@ -4,6 +4,10 @@
     <t t-name="mail.AttachmentPanel">
         <t t-set="title">Attachments</t>
         <ActionPanel title="title">
+            <div t-if="props.thread.model !== 'mail.box'" class="form-check form-switch">
+                <input t-on-change="toggleAllowPublicUpload" class="form-check-input" type="checkbox" role="switch" t-att-id="`im_livechat.allow_upload_${props.thread.uuid}`" t-att-checked="props.thread.allow_public_upload"/>
+                <label class="form-check-label" t-att-for="`im_livechat.allow_upload_${props.thread.uuid}`" t-esc="toggleAllowPublicUploadText "/>
```

Comment:
id trick only working on the assumption a single panel is shown for a thread at the same time, but I guess that's acceptable for now

Reply:
It's relying on `uuid` which should be unique by definition. Am I missing something?

Reply:
It's unique per thread but not per attachment panel. We currently only have a single attachment panel at the same time so that's not an issue, just pointing it out.
A safe way is just to increment a global counter for each new component setup.
Also I would really like we could completely get rid of uuid in business code at some point.